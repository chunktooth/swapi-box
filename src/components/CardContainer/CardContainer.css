.Card-container {
	display: flex;
	flex-wrap: wrap;
	justify-content: space-evenly;
	position: relative;
	top: 200px;
	box-sizing: border-box;
	width: 100%;
	margin: auto;
}