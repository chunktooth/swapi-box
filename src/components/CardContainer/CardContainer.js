import React from 'react';
import Card from '../Card/Card';
import './CardContainer.css';

const CardContainer = ({categories}) => {
	const displayCards = categories.map(data => {

		return <Card 
			key={data.name}
			categories={data} 
		/>
	});

	return (
		<section className="Card-container">
			{displayCards}
		</section>
	)
}

export default CardContainer;



