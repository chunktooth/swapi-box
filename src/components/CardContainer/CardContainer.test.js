import React from 'react';
import CardContainer from './CardContainer/CardContainer';
import {Shallow} from 'enzyme';

describe('CardContainer', () => {
	let container;

	beforeEach(() => {
		container = shallow(<CardContainer />);
	});

	it('should match the snapshot', () => {
		expect(container).toMatchSnapshot();
	});
	
})