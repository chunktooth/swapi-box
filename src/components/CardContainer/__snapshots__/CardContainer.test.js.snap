// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          exact={true}
          path="/people"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/planets"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/vehicles"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/favorites"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/people",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/planets",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/vehicles",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/favorites",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/people"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/planets"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/vehicles"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/favorites"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/people",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/planets",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/vehicles",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
